File: csharp/log-levels/LogLevels.cs
using System.Text.RegularExpressions;

static class LogLine
{
    public static string Message(string logLine) => logLine.Split(":")[1].Trim();
    public static string LogLevel(string logLine)
    {
        var logLevel = logLine.Split(":")[0].Trim();
        var rgx = new Regex(@"[\[\]]");
        return rgx.Replace(logLevel, "").ToLower();
    }
    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: clojure/log-levels/src/log_levels.clj
(ns log-levels
  (:require [clojure.string :as str]))

(def split-regex #"(?=\[.*\])*: ")

(defn split-log
  [log-line]
  (str/split log-line split-regex))

(defn message
  "Takes a string representing a log line
   and returns its message with whitespace trimmed."
  [log-line]
  (str/trim (get (split-log log-line) 1))
  )

(defn log-level
  "Takes a string representing a log line
   and returns its level in lower-case."
  [log-line]
  (str/lower-case (str/replace (get (split-log log-line) 0) #"[\[\]]" ""))
  )

(defn reformat
  "Takes a string representing a log line and formats it
   with the message first and the log level in parentheses."
  [log-line]
  (str (message log-line) " (" (log-level log-line) ")")
  )
